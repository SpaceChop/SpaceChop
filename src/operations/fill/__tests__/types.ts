import FillConfig, { FillConfig as Config } from '../types';

describe('should validate fill config', () => {
  describe('should throw on invalid config', () => {
    it('require width or height', () => {
      const config = {};
      expect(() => FillConfig.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('require valid width and height', () => {
      let config: Config = {
        width: -100,
        height: 100,
      };
      expect(() => FillConfig.check(config)).toThrowErrorMatchingSnapshot();

      config = {
        width: 100,
        height: -100,
      };
      expect(() => FillConfig.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('require valid gravity', () => {
      const config = {
        width: 100,
        height: 100,
        gravity: 'right',
      };
      expect(() => FillConfig.check(config)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('should now throw on valid config', () => {
    it('width & height', () => {
      const config: Config = {
        width: 100,
        height: 100,
      };
      expect(() => FillConfig.check(config)).not.toThrow();
    });

    it('width & height & gravity', () => {
      const config: Config = {
        width: 100,
        height: 100,
        gravity: 'center',
      };
      expect(() => FillConfig.check(config)).not.toThrow();
    });
  });
});
