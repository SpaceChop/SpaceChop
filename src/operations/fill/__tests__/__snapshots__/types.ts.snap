// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate fill config should throw on invalid config require valid gravity 1`] = `"Expected (\\"center\\" | \\"face\\" | \\"north\\" | \\"northeast\\" | \\"east\\" | \\"southeast\\" | \\"south\\" | \\"southwest\\" | \\"west\\" | \\"northwest\\") | { from_path: string; }, but was string"`;

exports[`should validate fill config should throw on invalid config require valid width and height 1`] = `"Expected number | { from_path: string; }, but was number"`;

exports[`should validate fill config should throw on invalid config require valid width and height 2`] = `"Expected number | { from_path: string; }, but was number"`;

exports[`should validate fill config should throw on invalid config require width or height 1`] = `"Expected number | { from_path: string; }, but was undefined"`;
